File: backend/app/schemas/project.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime

class ProjectBase(BaseModel):
    """Base schema for Project data"""
    name: str
    description: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

class ProjectCreate(ProjectBase):
    """Schema for creating a new project"""
    pass

class ProjectUpdate(ProjectBase):
    """Schema for updating a project"""
    name: Optional[str] = None

class Project(ProjectBase):
    """Schema for project response"""
    id: int
    user_id: int
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/project.ts
export interface Project {
    id: number;
    name: string;
    description: string;
    user_id: number;
    created_at: string;
    updated_at: string;
}

// Type for creating a new project
export interface ProjectCreate {
    name: string;
    description: string;
}

// Type for updating an existing project
export interface ProjectUpdate {
    name?: string;
    description?: string;
}